napari/_qt/tests/test_qt_play.py

```diff
@@ -0,0 +1,188 @@
+from contextlib import contextmanager
+
+import numpy as np
+import pytest
+
+from napari._qt.qt_viewer import QtViewer
+from napari.components import ViewerModel
+
+from ...components import Dims
+from ..qt_dims import AnimationThread
+
+
+# A lot of this code looks messy and unnecessary (and some of it may be!)
+# most of the weird bits are an attempt to circumvent thread timing
+# non-determinism on Cirrus CI OSX tests.
+# see https://github.com/napari/napari/pull/607 for more info
+@contextmanager
+def make_thread(
+    qtbot, nframes=8, fps=20, frame_range=None, playback_mode='loop'
+):
```

Comment:
I would love you to revisit after #650 . Possibly will make your life easier @tlambert03